server/views/pages/bookings/visit.njk

```diff
@@ -4,19 +4,47 @@
 {% from "govuk/components/button/macro.njk" import govukButton %}
 
 {% set pageTitle = "Visit booking details" %}
+{% set referenceType = "Booking reference" %}
+{% if isRequest or isRejected %}
+  {% set pageTitle = "Visit request details" %}
+{% endif %}
 
 {% block content %}
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
 
-      {# Messages #}
-      {% for message in messages %}
-        {{ mojAlert(message) }}
-      {% endfor %}
+      {% if isRequest or isRejected %}
+        {% set pageTitle = "Visit request details" %}
+        {% set referenceType = "Request reference" %}
+
+        {% if isRequest %}
+          {{ mojAlert({
+            variant: "information",
+            title: "Your request needs to be reviewed",
+            showTitleAsHeading: true,
+            dismissible: false,
+            text: "This visit is not booked yet. It needs to be checked by " + prison.prisonName + "."
+          }) }}
+        {% elif isRejected %}
+          {{ mojAlert({
+            variant: "information",
+            dismissible: false,
+            text: "This request was rejected by " + prison.prisonName + "."
+          }) }}
+        {% endif %}
+
+      {% else %}
+        {# Messages only shown on non-requested type visits #}
```

Comment:
A lot of nested logic in the template here. `messages` are supposed to be a set of `MoJ Alert(s)` that can be shown on the page. So might be simpler and easier to test to add them as appropriate in the controller (either directly or via a utility function).